### PR TITLE
Unhide DA.Internal.LF.coerceContractId

### PR DESCRIPTION
**Pull request by akrmn**

### CODE REVIEW

File: compiler/damlc/daml-stdlib-src/DA/Internal/LF.daml [hunk 1]
```diff
@@ -151,8 +151,9 @@ instance Show (ContractId a) where
     None -> "<contract-id>"
     Some t -> t
 
--- | HIDE This is currently used in the internals of Daml script and Daml triggers
--- but not really something that we want to expose to users.
+-- | Used to convert the type index of a `ContractId`, since they are just
+-- pointers. Note that subsequent fetches and exercises might fail if the
+-- template of the contract on the ledger doesn't match.
```
Comment (remyhaemmerle-da) [inline compiler/damlc/daml-stdlib-src/DA/Internal/LF.daml:156]: Maybe we could add a recommendation to use more type-safe function, like `fromInterface`, `toInterface` , `fromRequiringInterface`

**Review by remyhaemmerle-da — APPROVED** [review]
:disappointed:

**Comment by basvangijzel-DA** [comment]
What is the context for adding this back?

**Comment by akrmn** [comment]
> What is the context for adding this back?

this is not really adding it back, just adding it _to the docs_. The reason is that we were using it anyways, specifically in the code generated by `daml ledger export`.

**Pull request merged**